### ISSUE TITLE
💅 useNamingConvention should not be called if the type of the object is explicit

### ISSUE THREAD

**Issue by kiyoon** [biomejs/biome#7276]
### Environment information

```bash
My API is following snake_case. Thus, I have many interface definitions for use in the API calls.

It makes sense that the biome warns about the naming convention here, and I can ignore the lint in this file.


interface ApiParam {
  person_name: string
  date_of_birth: string
}


However, whenever I use this structure anywhere else, I get the same warning.


// The callAPI definition suggests that the param should be of type ApiParam, thus we can skip the warning here
await callAPI({person_name: "John", date_of_birth: "1999-01-01"})
// or, the user gave an explicit type. The linter can warn about the interface definition, but it's cluttering when warning about the usage.
let callParams: ApiParam = {person_name: "John", date_of_birth: "1999-01-01"}
// or
let callParams = {person_name: "John", date_of_birth: "1999-01-01"} as ApiParam
```

### Rule name

useNamingConvention

### Playground link

https://biomejs.dev/playground/?lintRules=useNamingConvention&enabledAssist=false&files.main.ts=aQBuAHQAZQByAGYAYQBjAGUAIABBAHAAaQBQAGEAcgBhAG0AIAB7AAoAIAAgAHAAZQByAHMAbwBuAF8AbgBhAG0AZQA6ACAAcwB0AHIAaQBuAGcACgAgACAAZABhAHQAZQBfAG8AZgBfAGIAaQByAHQAaAA6ACAAcwB0AHIAaQBuAGcACgB9AAoACgBhAHMAeQBuAGMAIABmAHUAbgBjAHQAaQBvAG4AIABjAGEAbABsAEEAUABJACgAcABhAHIAYQBtADoAIABBAHAAaQBQAGEAcgBhAG0AKQAgAHsACgAgACAALwAvACAAUwBpAG0AdQBsAGEAdABlACAAYQBuACAAQQBQAEkAIABjAGEAbABsAAoAfQAKAAoALwAvACAAVABoAGUAIABjAGEAbABsAEEAUABJACAAZABlAGYAaQBuAGkAdABpAG8AbgAgAHMAdQBnAGcAZQBzAHQAcwAgAHQAaABhAHQAIAB0AGgAZQAgAHAAYQByAGEAbQAgAHMAaABvAHUAbABkACAAYgBlACAAbwBmACAAdAB5AHAAZQAgAEEAcABpAFAAYQByAGEAbQAsACAAdABoAHUAcwAgAHcAZQAgAGMAYQBuACAAcwBrAGkAcAAgAHQAaABlACAAdwBhAHIAbgBpAG4AZwAgAGgAZQByAGUACgBhAHcAYQBpAHQAIABjAGEAbABsAEEAUABJACgAewAgAHAAZQByAHMAbwBuAF8AbgBhAG0AZQA6ACAAIgBKAG8AaABuACIALAAgAGQAYQB0AGUAXwBvAGYAXwBiAGkAcgB0AGgAOgAgACIAMQA5ADkAOQAtADAAMQAtADAAMQAiACAAfQApAAoALwAvACAAbwByACwAIAB0AGgAZQAgAHUAcwBlAHIAIABnAGEAdgBlACAAYQBuACAAZQB4AHAAbABpAGMAaQB0ACAAdAB5AHAAZQAuACAAVABoAGUAIABsAGkAbgB0AGUAcgAgAGMAYQBuACAAdwBhAHIAbgAgAGEAYgBvAHUAdAAgAHQAaABlACAAaQBuAHQAZQByAGYAYQBjAGUAIABkAGUAZgBpAG4AaQB0AGkAbwBuACwAIABiAHUAdAAgAGkAdAAnAHMAIABjAGwAdQB0AHQAZQByAGkAbgBnACAAdwBoAGUAbgAgAHcAYQByAG4AaQBuAGcAIABhAGIAbwB1AHQAIAB0AGgAZQAgAHUAcwBhAGcAZQAuAAoAbABlAHQAIABjAGEAbABsAFAAYQByAGEAbQBzADoAIABBAHAAaQBQAGEAcgBhAG0AIAA9ACAAewAgAHAAZQByAHMAbwBuAF8AbgBhAG0AZQA6ACAAIgBKAG8AaABuACIALAAgAGQAYQB0AGUAXwBvAGYAXwBiAGkAcgB0AGgAOgAgACIAMQA5ADkAOQAtADAAMQAtADAAMQAiACAAfQAKAC8ALwAgAG8AcgAKAGwAZQB0ACAAYwBhAGwAbABQAGEAcgBhAG0AcwAyACAAPQAgAHsACgAgACAAcABlAHIAcwBvAG4AXwBuAGEAbQBlADoAIAAiAEoAbwBoAG4AIgAsAAoAIAAgAGQAYQB0AGUAXwBvAGYAXwBiAGkAcgB0AGgAOgAgACIAMQA5ADkAOQAtADAAMQAtADAAMQAiACwACgB9ACAAYQBzACAAQQBwAGkAUABhAHIAYQBtAAoA

### Expected result

I believe it's more usable when the usage is not forced, and instead warn on the definition. Or, at least, if the definition has biome-ignore, the usage should also follow the ignore settings and not warn twice.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
